domain/machine/state/reboot_test.go

```diff
@@ -0,0 +1,230 @@
+	c.Assert(err, jc.ErrorIsNil)
+
+	// Verify: Check which machine needs reboot
+	isRebootNeeded, err := s.state.IsMachineRebootRequired(context.Background(), "a-l-i-ve")
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(isRebootNeeded, jc.IsFalse)
+	isRebootNeeded, err = s.state.IsMachineRebootRequired(context.Background(), "d-e-a-d")
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(isRebootNeeded, jc.IsTrue)
+}
+
+func (s *stateSuite) TestCancelMachineReboot(c *gc.C) {
+	// Setup: Create a machine with a given ID and add its ID to the reboot table.
+	err := s.state.CreateMachine(context.Background(), "", "", "u-u-i-d")
+	c.Assert(err, jc.ErrorIsNil)
+	err = s.runQuery(`INSERT INTO machine_requires_reboot (machine_uuid) VALUES ("u-u-i-d")`)
+	c.Assert(err, jc.ErrorIsNil)
+
+	// Call the function under test
+	err = s.state.CancelMachineReboot(context.Background(), "u-u-i-d")
+	c.Assert(err, jc.ErrorIsNil)
+
+	// Verify: Check if the machine needs reboot
+	isRebootNeeded, err := s.state.IsMachineRebootRequired(context.Background(), "u-u-i-d")
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(isRebootNeeded, jc.IsFalse)
+}
+
+func (s *stateSuite) TestCancelMachineRebootIdempotent(c *gc.C) {
+	// Setup: Create a machine with a given ID  add its ID to the reboot table.
+	err := s.state.CreateMachine(context.Background(), "", "", "u-u-i-d")
+	c.Assert(err, jc.ErrorIsNil)
+	err = s.runQuery(`INSERT INTO machine_requires_reboot (machine_uuid) VALUES ("u-u-i-d")`)
+	c.Assert(err, jc.ErrorIsNil)
+
+	// Call the function under test, twice (idempotency)
+	err = s.state.CancelMachineReboot(context.Background(), "u-u-i-d")
+	c.Assert(err, jc.ErrorIsNil)
+	err = s.state.CancelMachineReboot(context.Background(), "u-u-i-d")
+	c.Assert(err, jc.ErrorIsNil)
+
+	// Verify: Check if the machine needs reboot
+	isRebootNeeded, err := s.state.IsMachineRebootRequired(context.Background(), "u-u-i-d")
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(isRebootNeeded, jc.IsFalse)
+}
+
+func (s *stateSuite) TestCancelMachineRebootSeveralMachine(c *gc.C) {
+	// Setup: Create several machine with a given IDs,  add both ids in the reboot table
+	err := s.state.CreateMachine(context.Background(), "alive", "a-l-i-ve", "a-l-i-ve")
+	c.Assert(err, jc.ErrorIsNil)
+	err = s.state.CreateMachine(context.Background(), "dead", "d-e-a-d", "d-e-a-d")
+	c.Assert(err, jc.ErrorIsNil)
+	err = s.runQuery(`INSERT INTO machine_requires_reboot (machine_uuid) VALUES ("a-l-i-ve")`)
+	c.Assert(err, jc.ErrorIsNil)
+	err = s.runQuery(`INSERT INTO machine_requires_reboot (machine_uuid) VALUES ("d-e-a-d")`)
+	c.Assert(err, jc.ErrorIsNil)
+
+	// Call the function under test
+	err = s.state.CancelMachineReboot(context.Background(), "a-l-i-ve")
+	c.Assert(err, jc.ErrorIsNil)
+
+	// Verify: Check which machine needs reboot
+	isRebootNeeded, err := s.state.IsMachineRebootRequired(context.Background(), "a-l-i-ve")
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(isRebootNeeded, jc.IsFalse)
```

Comment:
Assertions fail the build, but optimistically we want to continue. So fail hard on errors or things that we must have, otherwise, we should use Check.

```suggestion
	c.Assert(err, jc.ErrorIsNil)
	c.Check(isRebootNeeded, jc.IsFalse)
```